### ISSUE TITLE
Race condition issue

### ISSUE THREAD

**Issue by opoudjis** [relaton/relaton#96]
Intermittently getting this error with cache, now that there are concurrent asynchronous fetches:

````
#<Thread:0x00007f9f25f2f578 /Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/workers_pool.rb:10 run> terminated with exception (report_on_exception is true):
/Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/db_cache.rb:111:in `delete': No such file or directory @ apply2files - /Users/nickn/.relaton/cache/ogc/ogc_15_104r5.redirect (Errno::ENOENT)
	from /Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/db_cache.rb:111:in `delete'
	from /Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/db.rb:378:in `check_bibliocache'
	from /Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/db.rb:73:in `fetch'
	from /Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/db.rb:108:in `block in fetch_async'
	from /Users/nickn/.rvm/gems/ruby-3.0.2/gems/relaton-1.9.4/lib/relaton/workers_pool.rb:11:in `block (2 levels) in initialize'
````

**Comment by andrew2net** [comment]
@opoudjis I've added `synchronize` blocks in places where relaton modifies files. Since I'm unable reproduce the issue could you pull the latest relaton, build gem locally and try it. If it works I'll release new version.

**Comment by opoudjis** [comment]
Thank you, I've run 10 times on the source file, and all have succeeded; before I was getting it crashing 2 times out of 3.

For reference, the bibliography of the document was:

````
[bibliography]
== Bibliography

* [[[bauman2021,1]]] Baumann, P., A _General Conceptual Framework for Multi-Dimensional Spatio-Temporal Data Sets_, Environmental Modelling and Software, 2021

* [[[gilber1948,2]]] Gilber, W., _A Cube-filling Hilbert Curve_, Mathematical Intelligencer 6(3), 1984

* [[[goodchild1983,3]]] Goodchild, M. F. and Grandfield, A. W., _Optimizing Raster Storage: An Examination of Four Alternatives,_ IAuto-Carto 6, 1983

* [[[hilton1960,4]]] Hilton, P. J. and Wylie, S., _Homology Theory, an Introduction to Algebraic Topology_, Cambridge University Press, 1960

* [[[rfc3986,IETF RFC 3986]]] IETF RFC 3986 Internet Engineering Task Force: Uniform Resource Identifier (URI): Generic Syntax, 2005. Available at: https://tools.ietf.org/html/rfc3986

* [[[epsg,6]]] International Association of Oil & Gas Producers, EPSG Data Set. Aavailable at: http://www.epsg.org/

* [[[iso19501,ISO/IEC 19501:2005]]] International Organization for Standardization /International Electrotechnical Commission, ISO/IEC 19501:2005, _Information technology ― Open Distributed Processing ― Unified Modeling Language (UML) Version 1.4.2_, Geneva, Switzerland, 2005

* [[[ISO_19144-1_2009,ISO 19144-1:2009]]] International Organization for Standardization, ISO 19144-1:2009 _Classification Systems -- Part 1: Classification system structure_, Geneva, Switzerland, 2009.

* [[[ISO_19144-2_2012,ISO 19144-2:2012]]] International Organization for Standardization, ISO 19144-2:2012 _Classification Systems -- Part 2: Land Cover Meta Language (LCML)_, Geneva, Switzerland, 2012.

* [[[ISO_19101-1_2014_2,ISO 19101-1:2014]]] International Organization for Standardization, ISO 19101-1:2014, _Geographic information __—_ _Reference model -- Part 1: Fundamentals,_ Geneva, Switzerland, 2014

* [[[ISO_19123-2_2018,ISO 19123-2:2018]]] ISO 19123-2:2018, Geographic information — Schema for coverage geometry and functions — Part 2: Coverage implementation schema

* [[[kidner2003,12]]] Kidner, D.B., _Higher-order interpolation of regular grid digital elevation models_, International Journal of Remote Sensing Vol. 24, No. 14, July 2003, pp. 2981-2987

* [[[laurini1992,13]]] Laurini, R. and Thompson, D., _Fundamentals of Spatial Information Systems,_ Academic Press, 1992

* [[[dictionary1996,14]]] n., _Dictionary of Computing_, Fourth Edition, Oxford University Press, 1996

* [[[object2005,OMG UML 2.0]]] Object Management Group, Unified Modeling Language, Version 2.0, Needham, Massachusetts, USA, 2005. Available at: https://www.omg.org/spec/UML/2.0/

* [[[OGC_compliance,16]]] Open Geospatial Consortium, Compliance Navigation. Available at: http://www.opengeospatial.org/compliance

* [[[OGC_09-146r6,OGC 09-146r6]]] Open Geospatial Consortium, _OGC Coverage Implementation Schema_, 09-146r6, Wayland, Massachusetts, USA, 2017. Available at: http://docs.opengeospatial.org/is/09-146r6/09-146r6.html

* [[[OGC_09-110r4,OGC 09-110r4]]] Open Geospatial Consortium, _The OGC(R) WCS (Web Coverage Service) 2.0 Interface Standard- Core_, 09-110r4, Wayland, Massachusetts, USA, 2102. Available at: http://www.opengeospatial.org/standards/wcs

* [[[ogc2007,OGC 07-011]]] Open Geospatial Consortium, _The OpenGIS Specification: Abstract Specification Topic 6_, Schema for coverage geometry and functions, Wayland, Massachusetts, USA, 2007. Available at: http://www.opengeospatial.org/docs/as

* [[[ogc2017,OGC 15-104r5]]] Open Geospatial Consortium, _The OpenGIS Specification: Abstract Specification Topic 21_, Discrete Global Grid Systems, Wayland, Massachusetts, USA, 2017. Available at: http://www.opengeospatial.org/docs/as

* [[[OGC_15-104r5,OGC 15-104r5]]] Open Geospatial Consortium, _Topic 21: Discrete Global Grid Systems Abstract Specification_, OGC 15-104r5, Wayland, Massachusetts, USA, 2017. Available at: https://www.ogc.org/docs/as
````

**Comment by andrew2net** [comment]
fixed in v 1.9.5
